Doc/library/asyncio-eventloop.rst

```diff
@@ -210,6 +210,44 @@ Running and stopping the loop
    .. versionchanged:: 3.12
       Added the *timeout* parameter.
 
+.. method:: loop.run_forever_setup()
+
+   Set up an event loop so that it is ready to start actively looping and
+   processing events.
+
+   Returns the state that must be restored when the loop concludes. This state
+   should be passed in as arguments to :meth:`loop.run_forever_cleanup()`.
+
+   .. note::
+
+      This method is only needed if you are writing your own ``EventLoop``
```

Comment:
Maybe this? Or maybe I misunderstand and the subclass would *call* this and the cleanup method?

```suggestion
      Overriding this method is only needed if you are writing your own ``EventLoop``
```

Reply:
The usage I'm envisaging (and the usage that would be needed by Toga would be to *call* the startup/cleanup methods, but provide a custom "while True: process event" loop that does whatever is needed for integration. 

Overriding may also be required - the Windows event loop does this. However, I suspect that's more a case of Python's implementation using subclassing to satisfy the needs of all platforms, rather than something you'll see with GUI integration.